tests/ut/ir/transforms/test_simplify_pass.py

```
@@ -621,6 +619,51 @@ def post_loop_operand(program):
+        The assignment must land inside the loop body: its RHS names the loop
+        variable, so it cannot be hoisted past the loop — and the last iteration
+        writing last is exactly the value a leak-mode post-loop read expects.
+        """
+
+        @pl.program
+        class Before:
+            @pl.function
+            def main(self, out: pl.Tensor[[8], pl.INDEX]):
+                acc_next: pl.Scalar[pl.INDEX] = 0
+                for k in pl.range(4):
+                    for j, (acc,) in pl.range(0, 1, init_values=(k,)):
```

Ruff (0.16.2)

[warning] Loop control variable `j` not used within loop body

Rename unused `j` to `_j`

(B007)